app/src/main/java/ru/practicum/android/diploma/ui/adapters/IndustryAdapter.kt

```diff
@@ -0,0 +1,52 @@
+package ru.practicum.android.diploma.ui.adapters
+
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import ru.practicum.android.diploma.databinding.IndustryListItemBinding
+import ru.practicum.android.diploma.domain.models.Industry
+import ru.practicum.android.diploma.ui.viewholders.IndustryViewHolder
+
+class IndustryAdapter(private val onItemClickListener: (Industry) -> Unit) :
+    RecyclerView.Adapter<IndustryViewHolder>() {
+
+    val listData = mutableListOf<Industry>()
+    private var selectedPosition: Int = -1
+
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ) = IndustryViewHolder(
+        IndustryListItemBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            false
+        )
+    ) {
+        selectedPosition = -1
+        onItemClickListener(listData[it])
+    }
+
+    override fun onBindViewHolder(holder: IndustryViewHolder, position: Int) {
+        holder.binding.industryRadioButton.isChecked = position == selectedPosition
+
+        holder.binding.industryRadioButton.setOnCheckedChangeListener { compoundButton, b ->
+            if (b) {
+                try {
+                    notifyItemChanged(selectedPosition)
+                } catch (e: IllegalStateException) {
+                    Log.d("IndustryAdapter", "Fail notifyItemChanged %s".format(e.toString()))
+                }
+                selectedPosition = holder.adapterPosition
```

Comment:
А тут не получится просто так:
```suggestion
                selectedPosition = position
```

Reply:
Студия сама ругается когда указываю явно position 
![Снимок](https://github.com/user-attachments/assets/fbbd4b01-c236-45d8-bb09-0d10eee7af3d)